### DISCUSSION TITLE
For nextcloud compat, when looking at relative links, Ignore ?fileID=123 the same way currently ignore #fileID=123

### DISCUSSION THREAD

**Discussion by mstyp** [gsantner/markor#1855] [Ideas] [answered]
I want to make relative links in nextcloud and have those relative links work in markor. (vise versa is another problem on nextclouds end[^1]). In nextcloud when you make a relative file link by default it adds `?fileId=123` to the end of the relative link, so when I try to edit that file in markor, it just makes an empty file called `filename.md?fileId=123`

When I brought up this issue in [nextcloud text app ](https://github.com/nextcloud/text/issues/2136), I was asked to:

>Could you also check if markor behaves differently for parameters provided in the fragment of an url (filename.md#fileId=123)?

Which I did, and in markor relative links with `#fileId=123` seems to work just.fine. Would it possible to make  `[herp](derp.md?fileId=123)` behavior the same as `[herp](derp.md#fileId=123)` in markor? 

I split my writing about evenly between my tablet and desktop and since I use nextcloud as my primary desktop text editor, its a big pain to my workflow when I open a quick reference note in markor and click a link, only to realize that woops I wrote this in nextcloud, gotta go in and edit the link so it works.

[^1]: its not really a fallback currently it does not work without it but they know bout that issue and have either fixed it or are working on it, cant tell tbh https://github.com/nextcloud/text/issues/2774

**Comment by gsantner** [comment, marked as answer]
@mstyp  
So you have some markdownfile.md, which contains some link `filename.md?someth=123` in it, and when you click that at markdownfile.md view-mode inside Markor it opens a file unwanted with the `?someth=123` postfix.

Is that right? Or what is the entrance point to Markor's bad `?` file?

**Reply by mstyp** [reply]
@gsantner yup that sums up the issue/ request/idea. clicking filename.md?someth=123 should just open filename.md, since everything after ? is just to help nextcloud find the file.

(sorry for double replying think I wrote in wrong feild initially)

**Comment by mstyp** [comment]
> Is that right?

@gsantner yup that sums up the issue/ request/idea. clicking `filename.md?someth=123` should just open `filename.md`, since everything after ? is just to help nextcloud find the file.

**Comment by gsantner** [comment]
OK I see. So there is one commit from today on the master branch changing one file. That's the same place where this can be implemented with one line of change :-).
